docs/en/operations/server-configuration-parameters/settings.md

```diff
@@ -69,29 +69,48 @@ If no conditions met for a data part, ClickHouse uses the `lz4` compression.
 </compression>
 ```
 
-<!--
 ## encryption {#server-settings-encryption}
 
-Configures a command to obtain a key to be used by [encryption codecs](../../sql-reference/statements/create/table.md#create-query-encryption-codecs). The command, or a shell script, is expected to write a Base64-encoded key of any length to the stdout.
+Configures a command to obtain a key to be used by [encryption codecs](../../sql-reference/statements/create/table.md#create-query-encryption-codecs). Key (or keys) should be written in enviroment variables or be set in configuration file.
+
+Keys can be hex or string. Their length must be equal to 16.
+
+!!! note "NOTE"
+    Storing keys in configuration file is not recommended. It isn't secure. You can move the keys into a separate config file on a secure disk and put a symlink to that config file to `config.d/` folder.
 
 **Example**
```

Comment:
I suggest reordering examples from easy to hard like the first example is about single key, immediately after that a warning that storing keys in config file is not recommended, the second example is about key in hex, then an example about a key from environment, and finally an example about multiple keys.